src/libcollections/btree/set.rs

```diff
@@ -74,53 +74,91 @@ pub struct BTreeSet<T> {
     map: BTreeMap<T, ()>,
 }
 
-/// An iterator over a BTreeSet's items.
+/// An iterator over a `BTreeSet`'s items.
+///
+/// This structure is created by the [`iter()`] method on [`BTreeSet`].
+///
+/// [`BTreeSet`]: struct.BTreeSet.html
+/// [`iter()`]: struct.BTreeSet.html#method.iter
 #[stable(feature = "rust1", since = "1.0.0")]
 pub struct Iter<'a, T: 'a> {
     iter: Keys<'a, T, ()>,
 }
 
-/// An owning iterator over a BTreeSet's items.
+/// An owning iterator over a `BTreeSet`'s items.
+///
+/// This structure is created by the [`into_iter()`] method on [`BTreeSet`]
+/// [`BTreeSet`] (provided by the [`IntoIterator`] trait).
```

Comment:
If one writes:

    ```rust
    [`BTreeSet`] (hello world)

    [`BTreeSet`]: https://google.com
    ```

rustdoc will treat this as a link to the url "https://hello world" which is not what we want. Instead, we have to write

    ```rust
    [`BTreeSet`][`BTreeSet`] (hello world)

    [`BTreeSet`]: https://google.com
    ```

which will link to https://google.com